test/hotspot/jtreg/runtime/ClassFile/OomWhileParsingRepeatedJsr.java

```diff
@@ -35,43 +35,44 @@
  * @bug 8016029
  * @requires vm.flagless
  * @library /test/lib
+ * @library /testlibrary/asm
  * @modules java.base/jdk.internal.misc
  *          java.desktop
  *          java.management
  * @run driver OomWhileParsingRepeatedJsr
  */
 
-import jdk.test.lib.JDKToolFinder;
-import jdk.test.lib.Platform;
 import jdk.test.lib.process.ProcessTools;
 import jdk.test.lib.process.OutputAnalyzer;
 
+import java.io.File;
+import java.nio.file.Files;
+
 public class OomWhileParsingRepeatedJsr {
 
     public static void main(String[] args) throws Exception {
 
-        // ======= Configure the test
-        String jarFile = System.getProperty("test.src") + "/testcase.jar";
         String className = "OOMCrashClass1960_2";
 
-        // ======= extract the test class
-        ProcessBuilder pb = new ProcessBuilder(new String[] {
-            JDKToolFinder.getJDKTool("jar"),
-            "xvf", jarFile } );
-        OutputAnalyzer output = new OutputAnalyzer(pb.start());
-        output.shouldHaveExitValue(0);
+        // create a file in the scratch dir
+        File classFile = new File(className + ".class");
+        classFile.createNewFile();
+
+        // fill it with the binary data of the class file
+        byte[] bytes = OOMCrashClass1960_2.dump();
```

Comment:
Okay so class loading triggers java (or jtreg) to compile the OOMCrashClass1960_2.java file then.